ci/build-webide.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+# Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+[ -z $1 ] && echo "usage build-webide [version] [latest:defaults true] [test:defaults true]" && exit 1
+
+echo "building webide image with $@"
+VERSION=$1
+LATEST=${2:-true}
+TEST=${3:-true}
+PUSH=${4:-true}
+DOCKER_FILE="$(dirname "${BASH_SOURCE[0]}")/../web-ide/ide-server/"
+
+fail() {
+    stopDocker
+    exit 1
+}
+
+ensureDockerStarted() {
+    local IMAGE_ID=$1
+    local limit=10
+    local i=0
+    local RESULT="START"
+    while [ $i -lt $limit -a $RESULT != "Up" ]; do
+        sleep 1
+        i=$(($i+1))
+        RESULT=$(docker container ls --filter=ancestor=$IMAGE_ID --format {{.Status}} | cut -c1-2)
+    done
+    if [ $i -eq $limit ]; then
+        echo "Could not start docker container: ls result='$RESULT'"
+        fail
+    fi
+}
+
+ensureWebideStarted() {
+    echo "ensuring endpoint is responsive"
+    local limit=10
+    local i=0
+    local RESULT=1
+    which curl > /dev/null 2>&1 #ensure dev-env does its thing before sending request
+    while [ $i -lt $limit -a $RESULT -ne 0 ]; do
+        sleep 1
+        i=$(($i+1))
+        curl -I http://localhost:8443
+        RESULT=$?
+    done
+    if [ $RESULT -ne 0 ]; then
+        echo "Could not start webide"
+        fail
+    fi
+}
+
+stopDocker() {
+    echo "stopping docker"
+    docker stop $(docker ps --format {{.ID}})
+}
+
+which docker > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+    echo "docker not installed!"
+    exit 1
+fi
+
+if [ "$LATEST" = true ]; then
+    docker build --rm --build-arg DAML_VERSION=$VERSION -t gcr.io/da-gcp-web-ide-project/daml-webide:$VERSION -t gcr.io/da-gcp-web-ide-project/daml-webide:latest $DOCKER_FILE
+else
+    docker build --rm --build-arg DAML_VERSION=$VERSION  -t gcr.io/da-gcp-web-ide-project/daml-webide:$VERSION $DOCKER_FILE
+fi
+
+if [ "$TEST" = true ]; then
+    IMAGE_ID=$(docker image ls --filter=reference="gcr.io/da-gcp-web-ide-project/daml-webide:$VERSION" --format "{{.ID}}")
+    echo "running container for image $IMAGE_ID"
+    docker run --rm -p 8443:8443 $IMAGE_ID > /dev/null 2>&1 &
```

Comment:
If you give a name to the container you are starting, you can then easily stop it by name. Also, why not just use the image name here?
```suggestion
    docker run --name daml-webide-test --rm -p 8443:8443 gcr.io/da-gcp-web-ide-project/daml-webide:$VERSION > /dev/null 2>&1 &
```

Reply:
Also, I'm not in a position to test right now, but doesn't `docker run` return the container ID? You could just store that in a variable if you're worried about name collisions.

Reply:
I've gotten to prefer image ids as they are definitive of the version of code I am running. I've added the container name